### PR TITLE
Issue #215  Update WebAuthn4J to 0.12.0 release

### PR DESCRIPTION
**Pull request by DTonoki**
## Analysis
Upstream WebAuthn4J stable release is 0.12.0 and OpenAM include release 0.9.14.

## Solution
Update WebAuthn4J to 0.12.0.

## Compatibility
OpenAM RP Server can use stored publickey with old 0.9.14 release.

## Testing
Registration(No resident key)

Registration(Resident key)

Authentication(No resident key)

Authentication(Resident key)

MFA Authentication

Upgrade OpenAM

### CODE REVIEW

File: openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java [hunk 1]
```diff
@@ -16,14 +16,25 @@
 
 package jp.co.osstech.openam.authentication.modules.webauthn;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java:23]: ObjectMapper is not used.

File: openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java [hunk 2]
```diff
@@ -54,6 +62,8 @@
         return ArrayUtil.clone(generatedChallenge.getValue());
     }
 
+    private WebAuthnManager webAuthnManager = WebAuthnManager.createNonStrictWebAuthnManager();
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java:65]: Please put instance variable definitions in one place.

File: openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java [hunk 3]
```diff
@@ -64,42 +74,61 @@ public WebAuthnAuthenticator validateCreateResponse(WebAuthnValidatorConfig conf
...
+            boolean _userVerificationRequired = config.isVerificationRequired(); 
+            boolean _userPresenceRequired = true; 
+            List<String> _expectedExtensionIds = Collections.emptyList(); //OpenAM desn't use extention now.
+
+            RegistrationRequest _registrationRequest = new RegistrationRequest(_attestationObjectBytes, _clientDataJsonBytes, 
+                    _clientExtensionJSON, _transports);
+            RegistrationParameters _registrationParameters = new RegistrationParameters(_serverProperty, _userVerificationRequired, 
+                    _userPresenceRequired, _expectedExtensionIds);
+
+            RegistrationData _registrationData;
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java:97]: I prefer the following code.

```
RegistrationData _registrationData;
try{
    _registrationData = webAuthnManager.parse(_registrationRequest);
    webAuthnManager.validate(_registrationData, _registrationParameters);
} catch (DataConversionException e){
    // If you would like to handle WebAuthn data structure parse error, please catch DataConversionException
    throw e;
} catch (ValidationException e){
    // If you would like to handle WebAuthn data validation error, please catch ValidationException
    throw e;
}
```

And it's the same on line 215.

**Comment by tsujiguchitky** [comment]
Please update copyright header too.

**Comment by DTonoki** [comment]
Thanks for your reviewing.
I got these.

File: openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java [hunk 4]
```diff
@@ -11,19 +11,29 @@
  * Header, with the fields enclosed by brackets [] replaced by your own identifying
  * information: "Portions copyright [year] [name of copyright owner]".
  *
- * Copyright 2019 Open Source Solution Technology Corporation
+ * Copyright 2020 Open Source Solution Technology Corporation
```
Comment (tsujiguchitky) [inline openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java:14]: `2019-2020` is correct.

Comment (DTonoki) [reply openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java]: got it sorry.

Comment (tsujiguchitky) [reply openam-authentication/openam-auth-webauthn/src/main/java/jp/co/osstech/openam/authentication/modules/webauthn/WebAuthn4JValidatorImpl.java]: I confirmed your fix, thanks.

**Comment by tsujiguchitky** [comment]
LGTM

**Review by tsujiguchitky — APPROVED** [review]

**Pull request merged**